processor_pool.go

```diff
@@ -30,6 +30,8 @@ type ProcessorPool struct {
 	processorsLock sync.Mutex
 	processors     []*Processor
 	processorsWG   sync.WaitGroup
+	pauseCount     int
+	pauseLock      *sync.Mutex
```

Comment:
Wat.  The methods take pointer receivers.  😐 https://godoc.org/sync#Mutex  ah whatteber, ripping it out anyway 😸